dotCMS/src/main/java/com/dotmarketing/portlets/contentlet/transform/strategy/DefaultTransformStrategy.java

```diff
@@ -0,0 +1,326 @@
+
+        if (binaries.isEmpty()) {
+            return;
+        }
+
+        // if we want to include binaries as they are (java.io.File) this is the flag you should turn on.
+        final boolean includeBinaries = options.contains(BINARIES);
+        for (final Field field : binaries) {
+            try {
+                final File conBinary = contentlet.getBinary(field.variable());
+                if (conBinary != null) {
+                    //If we want to see binaries. The binary-field per se must be replaced by file-name. We dont want to disclose any file specifics.
+                    if (includeBinaries) {
+
+                        if(conBinary.exists()) {
+                            final String dAPath = "/dA/%s/%s/%s";
+                            map.put(field.variable() + "Version", String.format(dAPath, contentlet.getInode(),field.variable(),conBinary.getName()));
+                            map.put(field.variable(), String.format(dAPath, contentlet.getIdentifier(),field.variable(),conBinary.getName()));
+                            map.put(field.variable() + "ContentAsset", contentlet.getIdentifier() + "/" +field.variable());
+                        }
+
+                    } else {
+                        //Otherwise lets just remove the binaries from the final map. as a precaution.
+                        map.remove(field.variable());
+                    }
+                }
+            } catch (IOException e) {
+                Logger.warn(this,
+                        "Unable to get Binary from field with var " + field.variable());
+            }
+        }
+    }
+
+    /**
+     * return categories as a list of key/values where the key is the categoryKey and the value is
+     * the categoryName
+     */
+    private void addCategories(final Contentlet contentlet, final Map<String, Object> map,
+            final Set<TransformOptions> options, final User user) {
+        final boolean allCategoriesInfo = options.contains(CATEGORIES_INFO);
+        final boolean includeCategoryName = options.contains(CATEGORIES_NAME);
+        if (includeCategoryName || allCategoriesInfo) {
+            try {
+                final List<Category> cats = toolBox.categoryAPI.getParents(contentlet, user, true);
+                final List<CategoryField> categoryFields = contentlet.getContentType()
+                        .fields(CategoryField.class).stream().filter(Objects::nonNull)
+                        .map(CategoryField.class::cast).collect(Collectors.toList());
+
+                for (final CategoryField categoryField : categoryFields) {
+                    final Category parentCategory = toolBox.categoryAPI
+                            .find(categoryField.values(), user, true);
+                    final List<Category> childCategories = new ArrayList<>();
+                    if (parentCategory != null) {
+                        for (Category category : cats) {
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Local variable 'category' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5596402)